python/cudf/cudf/core/_internals/timezones.py

```diff
@@ -111,7 +111,7 @@ def _find_ambiguous_and_nonexistent(
     tz_data_for_zone = get_tz_data(zone_name)
     transition_times = tz_data_for_zone["transition_times"]
     offsets = tz_data_for_zone["offsets"].astype(
-        f"timedelta64[{data._time_unit}]"
+        f"timedelta64[{data.time_unit}]"
```

Comment:
**suggestion** 
```suggestion
        np.timedelta64(0, np.datetime_data(data.dtype)).dtype
```
Rather than string-munging, WDYT?

Reply:
I find this a little more opaque than the existing code since we're creating a numpy scalar from which we only want the dtype. Maybe we should eventually invest in an `as_unit` method like pandas to do these resolution conversions https://pandas.pydata.org/docs/reference/api/pandas.DatetimeIndex.as_unit.html

Reply:
OK, let's leave as is for now.